waspc/src/Wasp/Psl/Ast/Model.hs

```diff
@@ -1,64 +1,21 @@
-{-# LANGUAGE DeriveDataTypeable #-}
-
 module Wasp.Psl.Ast.Model
   ( Model (..),
-    Body (..),
-    Element (..),
-    Field (..),
-    FieldType (..),
-    FieldTypeModifier (..),
     getFields,
   )
 where
 
-import Data.Data (Data)
-import Wasp.Psl.Ast.Attribute (Attribute)
 import Wasp.Psl.Ast.Common (Name)
-import Prelude hiding (Enum)
+import Wasp.Psl.Ast.Model.Common
+  ( Body (..),
+    Element (..),
+    Field,
+  )
 
 data Model
   = Model
       Name
       Body
   deriving (Show, Eq)
 
-newtype Body = Body [Element]
-  deriving (Show, Eq, Data)
-
-data Element
-  = ElementField Field
-  | ElementBlockAttribute Attribute
-  deriving (Show, Eq, Data)
-
--- TODO: To support attributes before the field,
---   we could just have `attrsBefore :: [[Attr]]`,
---   which represents lines, each one with list of attributes.
-data Field = Field
-  { _name :: String,
-    _type :: FieldType,
-    _typeModifiers :: [FieldTypeModifier],
-    _attrs :: [Attribute]
-  }
-  deriving (Show, Eq, Data)
-
-data FieldType
-  = String
-  | Boolean
-  | Int
-  | BigInt
-  | Float
-  | Decimal
-  | DateTime
-  | Json
-  | Bytes
-  | Unsupported String
-  | UserType String
-  deriving (Show, Eq, Data)
-
-data FieldTypeModifier
-  = List
-  | Optional
-  deriving (Show, Eq, Data)
-
 getFields :: Model -> [Field]
```

Comment:
Doesn't this function operate on Body really? It is not specific to model in any way, from what I see in its implementation. So why not move it to also to Common with the rest of the stuff?